engine-src/Game/LambdaHack/Client/UI/TutorialHints.hs

```diff
@@ -0,0 +1,30 @@
+module Game.LambdaHack.Client.UI.TutorialHints
+ (
+     TutorialHints (..)
+ )
+ where
+
+import Data.Text ()
+import Prelude 
+
+data TutorialHints =
+    DamageOfDifferentKind
+    | NewFloorNewOpportunity
+    | CannotHarmYouInMelee
+    | CaughtProjectile
+    | HitsWithNoDirectDamage
+    deriving(Enum, Eq, Bounded)
+
+instance Show TutorialHints where
+    show DamageOfDifferentKind =
+        "You took damage of a different kind than the normal piercing hit, which means your armor couldn't block any part of it. Normally, your HP (hit points, health) do not regenerate, so losing them is a big deal. Apply healing concoctions or take a long sleep to replenish your HP (but in this hectic environment not even uninterrupted resting that leads to sleep is easy)."
```

Comment:
Normally `show` and `read` should be inverses and `show` is used mostly for debugging, so perhaps it's better to define a separate function that does exactly what `show` does here.